src/components/button/button.test.js

```diff
@@ -210,4 +193,216 @@ context("Test for Button component", () => {
+    });
+
+    it("should pass accessibility tests for DashedButtonNoWrap", () => {
+      CypressMountWithProviders(<storiesDefault.DashedButtonNoWrap />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for DarkBackgroundButtonIconBefore", () => {
+      CypressMountWithProviders(<stories.DarkBackgroundButtonIconBefore />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for DarkBackgroundButtonIconAfter", () => {
+      CypressMountWithProviders(<stories.DarkBackgroundButtonIconAfter />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for DarkBackgroundButtonDisabled", () => {
+      CypressMountWithProviders(
+        <storiesDefault.DarkBackgroundButtonDisabled />
+      );
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for DarkBackgroundButtonFullWidth", () => {
+      CypressMountWithProviders(
+        <storiesDefault.DarkBackgroundButtonFullWidth />
+      );
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for DarkBackgroundButtonNoWrap", () => {
+      CypressMountWithProviders(<storiesDefault.DarkBackgroundButtonNoWrap />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for TertiaryButtonIconBefore", () => {
+      CypressMountWithProviders(<stories.TertiaryButtonIconBefore />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for TertiaryButtonIconAfter", () => {
+      CypressMountWithProviders(<stories.TertiaryButtonIconAfter />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for TertiaryButtonDestructive", () => {
+      CypressMountWithProviders(<storiesDefault.TertiaryButtonDestructive />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for TertiaryButtonDisabled", () => {
+      CypressMountWithProviders(<storiesDefault.TertiaryButtonDisabled />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for TertiaryButtonFullWitdth", () => {
+      CypressMountWithProviders(<storiesDefault.TertiaryButtonFullWitdth />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for TertiaryButtonFullWitdth", () => {
+      CypressMountWithProviders(<storiesDefault.TertiaryButtonNoWrap />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for fullWidth story", () => {
+      CypressMountWithProviders(<Button fullWidth>Foo</Button>);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for iconOnly story", () => {
+      CypressMountWithProviders(
+        <Button buttonType="primary" iconType="bin" aria-label="bin-icon" />
+      );
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for noWrap story", () => {
```

Comment:
Why is this story required when there are separate `*NoWrap` stories for each variant above?

Reply:
this is for all buttons :)